runtime/vara/src/lib.rs

```diff
@@ -0,0 +1,780 @@
+    type LengthToFee = IdentityFee<Balance>;
+    type FeeMultiplierUpdate = pallet_gear_payment::GearFeeMultiplier<Runtime, QueueLengthStep>;
+}
+
+impl_opaque_keys! {
+    pub struct SessionKeys {
+        pub babe: Babe,
+        pub grandpa: Grandpa,
+    }
+}
+parameter_types! {
+    pub const SelectedFraction: Percent = Percent::from_percent(25);
+}
+
+impl pallet_shift_session_manager::Config for Runtime {
+    type SelectedFraction = SelectedFraction;
+}
+
+impl pallet_session::Config for Runtime {
+    type Event = Event;
+    type ValidatorId = <Self as frame_system::Config>::AccountId;
+    type ValidatorIdOf = ();
+    type ShouldEndSession = Babe;
+    type NextSessionRotation = Babe;
+    type SessionManager = ShiftSessionManager;
+    type SessionHandler = <SessionKeys as OpaqueKeys>::KeyTypeIdProviders;
+    type Keys = SessionKeys;
+    type WeightInfo = pallet_session::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_sudo::Config for Runtime {
+    type Event = Event;
+    type Call = Call;
+}
+
+impl pallet_utility::Config for Runtime {
+    type Event = Event;
+    type Call = Call;
+    type WeightInfo = ();
+    type PalletsOrigin = OriginCaller;
+}
+
+pub struct GasConverter;
+impl gear_common::GasPrice for GasConverter {
+    type Balance = Balance;
+}
+
+impl pallet_gear_program::Config for Runtime {
+    type Event = Event;
+    type WeightInfo = pallet_gear_program::weights::GearProgramWeight<Runtime>;
+    type Currency = Balances;
+    type Messenger = GearMessenger;
+}
+
+parameter_types! {
+    pub const GasLimitMaxPercentage: Percent = Percent::from_percent(75);
+    pub BlockGasLimit: u64 = GasLimitMaxPercentage::get() * BlockWeights::get().max_block;
+    pub const WaitListTraversalInterval: u32 = 10;
+    pub const ExpirationDuration: u64 = MILLISECS_PER_BLOCK.saturating_mul(WaitListTraversalInterval::get() as u64);
+    pub const ExternalSubmitterRewardFraction: Perbill = Perbill::from_percent(10);
+    pub Schedule: pallet_gear::Schedule<Runtime> = Default::default();
+}
+
+impl pallet_gear::Config for Runtime {
+    type Event = Event;
+    type Currency = Balances;
+    type GasPrice = GasConverter;
+    type WeightInfo = pallet_gear::weights::GearWeight<Runtime>;
+    type Schedule = Schedule;
+    type OutgoingLimit = ConstU32<1024>;
+    type DebugInfo = DebugInfo;
+    type CodeStorage = GearProgram;
+    type MailboxThreshold = ConstU64<3000>;
+    type Messenger = GearMessenger;
+    type GasProvider = GearGas;
+    type BlockLimiter = GearGas;
+    type Scheduler = GearScheduler;
+}
+
+#[cfg(feature = "debug-mode")]
+impl pallet_gear_debug::Config for Runtime {
+    type Event = Event;
+    type WeightInfo = pallet_gear_debug::weights::GearSupportWeight<Runtime>;
+    type CodeStorage = GearProgram;
+    type Messenger = GearMessenger;
+}
+
+impl pallet_gear_scheduler::Config for Runtime {
+    type BlockLimiter = GearGas;
+    type ReserveThreshold = ConstU32<1>;
```

Comment:
Such things may be also be stored in primitives as new best practice:
```suggestion
    type ReserveThreshold = ConstU32<gear_primitives::ReserveThreshold>;
```

Reply:
This constant can be runtime-dependent, that is differ from runtime to runtime. I would suggest to define it in the `constants` module instead.
In primitives we collect some very basic generic concepts used in definitions of higher-level apis. Here I'm not inventing anything new - just copied the way it's done in Substrate and other chains.

Reply:
Yeah, I understand, but atm our constants for different runtimes are same and I see no reason to copy-paste and define them every time. But the `default_constants` mod seems ok for me.
I'll left it up to you so.

Reply:
Moved as much of the common stuff from both runtimes as possible in a separate crate: `gear-runtime-common`. This reduces "copy-pasting" to the minimum.